src/biconnectivity/articulation.jl

```diff
@@ -0,0 +1,55 @@
+"""
+Computes the articulation points(https://en.wikipedia.org/wiki/Biconnected_component)
+of a connected graph `g` and returns an array containing all cut vertices.
+"""
+function articulation(g::SimpleGraph) :: AbstractArray{Int}
+    state = Articulations(g)
+    for u in vertices(g)
+        state.depth[u] == 0 && visit!(state, g, u, u)
```

Comment:
I'm ok with this but we seem to be getting close to the line we drew a while back:

> Style point: prefer the short circuiting conditional over if/else when convenient, and where state is not explicitly being mutated (e.g., condition && error("message") is good; condition && i += 1 is not).